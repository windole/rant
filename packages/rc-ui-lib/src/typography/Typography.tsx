import React, { useContext } from 'react';
import classnames from 'classnames';
import { TypographyBaseProps } from './PropsType';
import ConfigProviderContext from '../config-provider/ConfigProviderContext';

const TypographyBase: React.FC<TypographyBaseProps & { renderType: string }> = ({
  type,
  size = 'md',
  level,
  center,
  ellipsis,
  className,
  children,
  strong,
  underline,
  disabled,
  renderType,
  delete: del,
  ...props
}) => {
  const { prefixCls, createNamespace } = useContext(ConfigProviderContext);
  const [bem] = createNamespace('typography', prefixCls);

  const elli = ellipsis === true ? 1 : ellipsis;
  return (
    <div
      className={classnames(
        className,
        bem([
          type,
          size,
          renderType,
          {
            center,
            strong,
            underline,
            disabled,
            delete: del,
            [`l${level}`]: renderType === 'title' && level,
          },
        ]),
        {
          'rc-ellipsis': elli === 1,
          [`rc-multi-ellipsis--l${elli}`]: elli && elli > 1,
        },
      )}
      {...props}
    >
      {children}
    </div>
  );
};

export default TypographyBase;
