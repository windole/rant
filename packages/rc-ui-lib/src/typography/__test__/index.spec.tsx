import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Typography from '..';

describe('Typography', () => {
  let wrapper;
  afterEach(() => {
    wrapper.unmount();
    jest.restoreAllMocks();
  });

  it('render correctly', async () => {
    wrapper = mount(<Typography.Text>这是一条文本</Typography.Text>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('render correctly when using type prop', async () => {
    wrapper = mount(<Typography.Text type="danger">这是一条文本</Typography.Text>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('render correctly when using underline prop', async () => {
    wrapper = mount(<Typography.Text underline>这是一条文本</Typography.Text>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('render correctly when using ellipsis prop', async () => {
    wrapper = mount(
      <Typography.Text ellipsis>
        In the process of internal desktop applications development, many different design specs and
        implementations would be involved
      </Typography.Text>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('render correctly when ellipsis has value', async () => {
    wrapper = mount(
      <Typography.Text ellipsis={2}>
        {' '}
        In the process of internal desktop applications development, many different design specs and
        implementations would be involved
      </Typography.Text>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('render correctly when using level prop', async () => {
    wrapper = mount(<Typography.Title level={1}>一级测试标题</Typography.Title>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
