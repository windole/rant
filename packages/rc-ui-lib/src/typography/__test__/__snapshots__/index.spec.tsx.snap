// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography render correctly 1`] = `
<Text>
  <TypographyBase
    renderType="text"
  >
    <div
      className="rc-typography rc-typography--md rc-typography--text"
    >
      这是一条文本
    </div>
  </TypographyBase>
</Text>
`;

exports[`Typography render correctly when ellipsis has value 1`] = `
<Text
  ellipsis={2}
>
  <TypographyBase
    ellipsis={2}
    renderType="text"
  >
    <div
      className="rc-typography rc-typography--md rc-typography--text rc-multi-ellipsis--l2"
    >
       
      In the process of internal desktop applications development, many different design specs and implementations would be involved
    </div>
  </TypographyBase>
</Text>
`;

exports[`Typography render correctly when using ellipsis prop 1`] = `
<Text
  ellipsis={true}
>
  <TypographyBase
    ellipsis={true}
    renderType="text"
  >
    <div
      className="rc-typography rc-typography--md rc-typography--text rc-ellipsis"
    >
      In the process of internal desktop applications development, many different design specs and implementations would be involved
    </div>
  </TypographyBase>
</Text>
`;

exports[`Typography render correctly when using level prop 1`] = `
<Title
  level={1}
>
  <TypographyBase
    level={1}
    renderType="title"
  >
    <div
      className="rc-typography rc-typography--md rc-typography--title rc-typography--l1"
    >
      一级测试标题
    </div>
  </TypographyBase>
</Title>
`;

exports[`Typography render correctly when using type prop 1`] = `
<Text
  type="danger"
>
  <TypographyBase
    renderType="text"
    type="danger"
  >
    <div
      className="rc-typography rc-typography--danger rc-typography--md rc-typography--text"
    >
      这是一条文本
    </div>
  </TypographyBase>
</Text>
`;

exports[`Typography render correctly when using underline prop 1`] = `
<Text
  underline={true}
>
  <TypographyBase
    renderType="text"
    underline={true}
  >
    <div
      className="rc-typography rc-typography--md rc-typography--text rc-typography--underline"
    >
      这是一条文本
    </div>
  </TypographyBase>
</Text>
`;
